ConsoleApp1/InternetShop.cs

```diff
@@ -0,0 +1,38 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace ConsoleApp1
+{
+    internal class InternetShop : Shop
```

Comment:
regarding your UML diagram, the shop is a derived class and InternetShop is the base so make adjust or scheme or code